builtin/providers/aws/resource_aws_s3_bucket_object.go

```diff
@@ -144,7 +148,10 @@ func resourceAwsS3BucketObjectPut(d *schema.ResourceData, meta interface{}) erro
 		return fmt.Errorf("Error putting object in S3 bucket (%s): %s", bucket, err)
 	}
 
-	d.Set("etag", resp.ETag)
+	// See https://forums.aws.amazon.com/thread.jspa?threadID=44003
+	d.Set("etag", strings.Trim(*resp.ETag, `"`))
+
+	d.Set("version_id", resp.VersionId)
 	d.SetId(key)
 	return nil
```

Comment:
I don't mind that it's not in place - if you feel it's good then i am good with it